src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +36,343 @@
+                                                           host_name_list)
+            except Exception as error:
+                message = "Exception {}: {}".format(type(error).__name__,
+                                                    str(error))
+                logger.error("{}".format(message))
+
+            # Create aggregate for hugepages configuration
+            try:
+                if enable_hugepages:
+                    aggregate_name = "HugePage_Aggr"
+                    aggregate_meta = [{'hugepages': 'True'}]
+                    config_oc_obj.configure_dell_aggregate(aggregate_name,
+                                                           aggregate_meta,
+                                                           host_name_list)
+            except Exception as error:
+                message = "Exception {}: {}".format(type(error).__name__,
+                                                    str(error))
+                logger.error("{}".format(message))
+
+            # Create normal aggregate
+            aggregate_name = "Normal_Aggr"
+            aggregate_meta = [{'pinned': 'False'}, {'hugepages': 'False'}]
+            config_oc_obj.configure_dell_aggregate(aggregate_name,
+                                                   aggregate_meta)
+    except Exception as error:
+        message = "Exception {}: {}".format(type(error).__name__, str(error))
+        logger.error("{}".format(message))
+
+
+def get_dell_compute_nodes_uuids():
+    logger.info("Getting dell compute node uuids")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    # Create nova client object
+    nova = nova_client.Client(2, **kwargs)
+
+    # Create the ConfigOvercloud object
+    config_oc_obj = ConfigOvercloud(nova)
+    dell_compute_nodes_uuid = config_oc_obj.get_dell_compute_nodes_uuid()
+    logger.debug("dell_compute_nodes_uuid: {}".format(dell_compute_nodes_uuid))
+    return dell_compute_nodes_uuid
+
+
+def reboot_compute_nodes():
+    logger.info("Rebooting dell compute nodes")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    n_client = nova_client.Client(2, **kwargs)
+    for compute in n_client.servers.list(detailed=False,
+                                         search_opts={'name': 'compute'}):
+        logger.info("Rebooting server with id: ".format(compute_name))
+        n_client.servers.reboot(compute.id)
+
+        # Wait for 5 mins to let dell nodes boot up
```

Comment:
As written, this code will reboot a compute node and then sleep 5 minutes, repeat.  This means that the total sleep time will be number_of_compute_nodes*5 minutes.  I think what we want to do is reboot all the compute nodes, then sleep 5 minutes, so I think we just need to unindent the above line and the following line 4 spaces.

Reply:
Agreed - thread it up , and rather than sleep 5 mns , wait for the computes to come back up , you should find some examples in existing code how to do that.

Reply:
Agreed that sleeping isn't the best way to do this.  A better way would be to wait until you can SSH into each of the computes.  You can do this with or without threads depending on how you want to do it.  I'm happy to leave this to a follow-on PR though.

Reply:
Added a ssh validation check to the reboot_compute_nodes method